docs/current_docs/manuals/developer/cookbook/snippets/copy-filter-dir/typescript/index.ts

```diff
@@ -0,0 +1,34 @@
+import { dag, Container, Directory, object, func } from "@dagger.io/dagger"
+
+@object()
+class MyModule {
+  /**
+   * Return a container with a filtered directory
+   */
+  @func()
+  copyDirectoryWithExclusions(
+    /**
+     * Source directory
+     */
+    source: Directory,
+    /**
+     * Directory exclusion pattern
+     */
+    excludeDirectory?: string,
+    /**
+     * File exclusion pattern
+     */
+    excludeFile?: string,
+  ): Container {
+    let filteredSource = source
+    if (excludeDirectory != null) {
+      filteredSource = filteredSource.withoutDirectory(excludeDirectory)
+    }
+    if (excludeFile != null) {
```

Comment:
You should usse `!` to also test `undefined`

```suggestion
    if (!excludeFile) {
```